asteroidsyncservice/watch.cpp

```diff
@@ -20,12 +20,24 @@
 #include "watch.h"
 
 #include <QDBusArgument>
+#include <QDBusConnection>
 #include <QDebug>
 
+#define SYNCSERVICE_SERVICE QStringLiteral("org.asteroidsyncservice")
+#define SYNCSERVICE_WATCH_PATH QStringLiteral("/org/asteroidsyncservice/Watch")
+#define SYNCSERVICE_WATCH_INTERFACE QStringLiteral("org.asteroidsyncservice.Watch")
+
 Watch::Watch(const QDBusObjectPath &path, QObject *parent) : QObject(parent), m_path(path)
 {
     m_iface = new QDBusInterface("org.asteroidsyncservice", path.path(), "org.asteroidsyncservice.Watch", QDBusConnection::sessionBus(), this);
 
+    m_watcher = new QDBusServiceWatcher(SYNCSERVICE_SERVICE, QDBusConnection::sessionBus(), QDBusServiceWatcher::WatchForOwnerChange, this);
+    QDBusConnection::sessionBus().connect(SYNCSERVICE_SERVICE, SYNCSERVICE_WATCH_PATH, SYNCSERVICE_WATCH_INTERFACE, "TimeServiceReady", this, SLOT(tServiceReady()));
+
+    connect(m_watcher, &QDBusServiceWatcher::serviceRegistered, [this]() {
+        QDBusConnection::sessionBus().connect(SYNCSERVICE_SERVICE, SYNCSERVICE_WATCH_PATH, SYNCSERVICE_WATCH_INTERFACE, "TimeServiceReady", this, SLOT(tServiceReady()));
```

Comment:
Cf related comment in asteroidsyncserviced: Why don't you just get a signal from m_iface here?

Reply:
m_iface works fine, is changed.